### ISSUE TITLE
java.lang.SecurityException: agent may not write

### ISSUE THREAD

**Issue by JanC** [jenkinsci/cobertura-plugin#54]
Hi,
I'm using the cobertura plugin in a job that runs on a Jenkins node. The jobs fails because

```
[Cobertura] Publishing Cobertura coverage report...
Publishing Cobertura coverage results...
Cobertura coverage report found.
ERROR: Build step failed with exception
java.lang.SecurityException: agent may not write 
```

I set up the Slave to master to `allow all < BUILDDIR >/.*` as described here but it still fails
https://wiki.jenkins-ci.org/display/JENKINS/Slave+To+Master+Access+Control

How can I set the correct rights for the plugin?

cheers

**Comment by udaysharma** [comment]
+1

Experiencing the same issue. Any guidance? It looks like the plugin is mixing slave (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/7.3.0/include) and master (/opt/bitnami/apps) paths.

Jenkins v2.68, Cobertura Plugin v1.10

```
Error:
agent may not mkdirs 

/opt/bitnami/apps/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/7.3.0/include

See https://jenkins.io/redirect/security-144 for more details
```

**Comment by jeffpearce** [comment]
@JanC @udaysharma - going through some older issues. Is this still a problem?

**Comment by masterhampel** [comment]
@jeffpearce Yes, I think this problem isn't fixed yet. I got this error too.

**Comment by JanC** [comment]
Hi,
I stopped using the plugin as I use Jenkins pipelines. I got it working correctly though after updating to a more recent version.

**Comment by basava12345** [comment]
Jenkins master is now more strict about what commands its slaves can send to the master. Unfortunately, this prevents some plugins from functioning correctly, as those plugins do not specify which commands are open for slaves to execute and which ones are not
For more reference
https://wiki.jenkins.io/display/JENKINS/Slave+To+Master+Access+Control

Here I feel fix should be mainly related to implementing the SECURITY-144-compat module in the plugin code.
However we have passing scenario also for the jobs running in slaves.
(Are the commands being used in this case are different ? )
